src/main/scala/io/picnicml/doddlemodel/cluster/DBSCAN.scala

```diff
@@ -0,0 +1,98 @@
+package io.picnicml.doddlemodel.cluster
+
+import breeze.linalg.functions.euclideanDistance
+import cats.syntax.option._
+import io.picnicml.doddlemodel.data.Features
+import io.picnicml.doddlemodel.typeclasses.Clusterer
+
+/** An immutable DBSCAN model.
+  *
+  * @param eps: the maximum distance between points in a group
+  * @param min_samples: the minimum number of point in a core group
+  *
+  * Examples:
+  * val model = DBSCAN()
+  * val model = DBSCAN(eps = 1.5)
+  * val model = DBSCAN(min_samples = 3)
+  * val model = DBSCAN(eps = 2.0, min_samples = 3)
+  */
+case class DBSCAN private(eps: Double, min_samples: Int,
+  private val label: Option[Array[Int]]) {}
```

Comment:
Some variables were not in camelCase because I thought this project wants to simulate sklearn.
But it makes sense that the consistency in a package is more important.
I will change to `minSamples` and `fitPredict`.